src/python/pants/rules/core/filedeps.py

```diff
@@ -16,24 +17,36 @@ class Filedeps(LineOriented, Goal):
   closure of targets are also included.
   """
 
-  # TODO: Until this implements more of the options of `filedeps`, it can't claim the name!
+  # TODO: Until this implements `--globs`, this can't claim the name `filedeps`!
   name = 'fast-filedeps'
 
+  @classmethod
+  def register_options(cls, register):
+    super().register_options(register)
+    register(
+      '--absolute', type=bool, default=True,
+      help='If True, output with absolute path; else, output with path relative to the build root.'
+    )
+
 
 @console_rule(Filedeps, [Console, Filedeps.Options, TransitiveHydratedTargets])
 def file_deps(console, filedeps_options, transitive_hydrated_targets):
 
-  uniq_set = OrderedSet()
+  absolute = filedeps_options.values.absolute
+
+  unique_rel_paths = set()
+  build_root = get_buildroot()
```

Comment:
So... I think that this is fine. But we should consider whether we want to DI the buildroot the same way we DI most everything in v2. The easiest thing would probably be to make it a singleton at engine-creation time: https://github.com/pantsbuild/pants/blob/da0989f9f00cf2a9eca58410a721d23a1fdf72bb/src/python/pants/init/engine_initializer.py#L329-L339

As you've seen, having this be a global mutable variable can be painful from a testing perspective, and we should look for opportunities to minimize that pain in v2.

Reply:
Hm, this sounds good to me but I don't really know how to approach this. Should we add a TODO or would someone have a few minutes to add a commit with this? I can try with a bit more direction also.

Reply:
AFAICT, the sum total of code involved would be:
```
 @rule(BuildRoot, []) 
 def build_root_singleton(): 
   return BuildRoot.instance
  # ... a few lines down in that file
  rules += [build_root_singleton]

# in a rule:
@rule(.., [.., BuildRoot])
```

This is not a big deal either way though.

Reply:
I was kind of hoping we'd just deprecate the absolute flag, rather than need to worry about this. WDYT?

Reply:
Implemented the rule Stu proposed. Don't want to wait for the deprecation cycle and I think this rule could be helpful in the future.

Thanks for the suggestion!